### ISSUE TITLE
fail to build on macos with libxmlsec1 1.2.32

### ISSUE THREAD

**Issue by kazzmir** [mehcode/python-xmlsec#169]
python-xmlsec fails to build on macos (with xcode 12.4) with libxmlsec1 1.2.32 installed. With libxmlsec 1.2.31 installed, python-xmlsec builds correctly.

The issue is that python-xmlsec has a check to see if the version libxmlsec1 is new enough, and if so then to set XMLSEC_NO_XKMS to 1. The problem is the version check is broken due to the subminor version of libxmlsec1 being bumped to 32. When the version check fails, then src/constants.c has this line
```
PYXMLSEC_ADD_NS_CONSTANT(XkmsNs, "XKMS");
```
Where this symbol is not defined anywhere, it was removed in a previous issue: https://github.com/mehcode/python-xmlsec/issues/26

In src/platform.h is the following line of code
```
#define XMLSEC_VERSION_HEX ((XMLSEC_VERSION_MAJOR << 8) | (XMLSEC_VERSION_MINOR << 4) | (XMLSEC_VERSION_SUBMINOR))
```
This allows for N bits in the major version (where N is reasonably large), 4 bits for the minor version, and 4 bits for the subminor version. Visually:
```
aaaaabbbbcccc
```
Where a's = the bits for the major version, b's = the bits for the minor version, and c's = the bits for the subminor version. Note that a subminor version greater than 15 already does not fit into the c bits, because 16 would be 5 bits (10000). This turns out to work currently because the current minor version is 2, thus the minor bits are '10'. When the minor bits are shifted left by 4 and binary or'd with the subminor bits the result is '111111' = 63 decimal. Now that the subminor version is 32, which is 100000, this conflicts with the minor version. The math ends up being '32 | 32' which is just 32.

To sum up: for libxmlsec1 1.2.31 the XMLSEC_VERSION_HEX value is (1<<8) | (2<<4) | (31) = 0x13f
For libxmlsec1 1.2.32 the value is (1<<8) | (2<<4) | 32 = 0x120.

Thus for 1.2.32 the following condition is false.
```
#if  XMLSEC_VERSION_HEX > 0x134
#define XMLSEC_NO_XKMS 1
#endif
```

A simple solution would be to increase the number of bits for each part of the version. Maybe 6 or 7 bits is enough. That would be
```
#define XMLSEC_VERSION_HEX ((XMLSEC_VERSION_MAJOR << 12) | (XMLSEC_VERSION_MINOR << 6) | (XMLSEC_VERSION_SUBMINOR))
```

For now a hacky workaround I am using is to modify libxmlsec's verison.h to declare its version as 1.2.31. This enables python-xmlsec to build.
```
/usr/local/Cellar/libxmlsec1/1.2.32/include/xmlsec1/xmlsec/version.h
#define XMLSEC_VERSION                  "1.2.31"
#define XMLSEC_VERSION_SUBMINOR         31
#define XMLSEC_VERSION_INFO             "3:31:2"
```
I am not sure if changing the header file will lead to other issues down the road.

**Comment by kazzmir** [comment]
My mistake, I see this is fixed in master already. It may be fixed in pypi version 1.3.10 but I can't seem to install that with pip for python2 (yes I know that is painfully old).

**Comment by kazzmir** [comment]
ce45868701220368f156114978c784cc5200d80c this was the commit that fixed it
